gems/aws-sdk-core/lib/aws-sdk-core/ecs_credentials.rb

```diff
@@ -52,17 +56,27 @@ class Non200Response < RuntimeError; end
     #   credentials are refreshed. `before_refresh` is called
     #   with an instance of this object when
     #   AWS credentials are required and need to be refreshed.
-    def initialize options = {}
+    def initialize(options = {})
       @retries = options[:retries] || 5
-      @ip_address = options[:ip_address] || '169.254.170.2'
+      @host = options[:ip_address] || '169.254.170.2'
```

Comment:
It should raise an error before line 95 right? How do you recommend setting these once? I think what I have is least confusing because we still need to check full uri is given anyway, so I do it once after defaults.

Reply:
Theres basically only two (non-error) cases right?  Either a credential_path is configured, in which case we use host and port from the options OR endpoint is configured, in which case we set the host and port from that so something like:
```
if @credential_path
  @port = options[:port] || 80
  ...
elsif endpoint
   override_full_uri
else
  raise ArgumentError
end
```

Reply:
I think that could work. If they're both configured, full URI is supposed to be ignored. That way does set them multiple times like I already am though.

Reply:
How does that set them multiple times? 

But anyway - just a suggestion - I'm fine with either way, just found the current initialization difficult to follow.

Reply:
Your snippet uses override_full_uri. But I see what you wanted to convey